torch/testing/_internal/common_device_type.py

```diff
@@ -344,6 +344,11 @@ def run(self, result=None):
             result.stop()
 
 
+# All available device types, this list must match
+# the derived *TestBase classes below
+ALL_AVAILABLE_DEVICE_TYPES = ['meta', 'cpu', 'cuda', 'xla']
```

Comment:
There's no list of all available device types because 3rd parties can programmatically extend the bases (like XLA does).

Reply:
oh... I see. then this approach is problematic.